chainer/functions/gaussian.py

```diff
@@ -0,0 +1,73 @@
+import numpy
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+class Gaussian(function.Function):
+
+    """Gaussian sampling function.
+
+    It takes mean and log(variance) as input and output sample drawn from
+    gaussian N(mean, variance).
+
+    """
+
+    def __init__(self):
+        self.eps = None
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 2)
+
+        m_type, v_type = in_types
+        type_check.expect(
+            m_type.dtype == numpy.float32,
+            v_type.dtype == numpy.float32,
+            m_type.shape == v_type.shape,
+        )
+
+    def check_type_fbackward(self, in_types, out_types):
```

Comment:
Typo `check_type_fbackward` . Be cautious that this function is not tested by unittests.